dashboard/admin.py

```diff
@@ -1,26 +1,35 @@
 from django.contrib import admin
+from django import forms
 from django.conf import settings
 from django.utils.safestring import mark_safe
 from django.template.defaultfilters import linebreaksbr
 
 from dashboard.common.db_util import canvas_id_to_incremented_id
 from .models import CourseViewOption, Course
 
-class CourseInline(admin.TabularInline):
-    model = Course
-
 class CourseViewOptionInline(admin.StackedInline):
     model = CourseViewOption
-
     exclude = ()
 
     # exclude disabled views
     for view in CourseViewOption.VIEWS:
         if view in settings.VIEWS_DISABLED:
             exclude += (view,)
 
+class CourseForm(forms.ModelForm):
+    class Meta:
+        model = Course
+        exclude = ()
+
+    def clean(self):
+        canvas_id = self.cleaned_data.get('canvas_id')
+        if not canvas_id.isdigit():
+            raise forms.ValidationError("Canvas ID must be an integer value")
```

Comment:
I think we should log the course_id so the message should say 
```suggestion
            raise forms.ValidationError(f"Course ID {canvas_id} must be an integer value")
```

Reply:
Thanks! Updated.